typing/ctype.ml

```diff
@@ -2251,18 +2251,18 @@ let is_principal ty =
 
 type unwrapped_type_expr =
   { ty : type_expr
-  ; modality : Mode.Modality.Const.t }
+  ; modality : Mode.Modality.Const.t
+  ; or_null : type_declaration option }
```

Comment:
Maybe leave a small comment motivating why this contains a `type_declaration` instead of a `bool`?